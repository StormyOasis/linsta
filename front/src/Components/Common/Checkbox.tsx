import React from 'react';
import styled from 'styled-components';
import { Flex } from './CombinedStyling';
import { CheckSVG } from './Icon';

const CheckboxWrapper = styled(Flex)`
    align-items: center;    
`;

const CheckboxInput = styled.input`    
    opacity: 0;
    width: 0;
    height: 0;
`;

const CheckboxLabel = styled.label<{ $isChecked: boolean, $width: string, $height: string }>`
    align-items: center;
    display: flex;
    position: relative;
    cursor: pointer;    
    justify-content: center;    
    width: ${props => props.$width};
    height: ${props => props.$height};
    border-radius: 50%;
    border: 2px solid ${props => props.theme['colors'].borderDefaultColor};    
    color: ${props => props.$isChecked ? props.theme['colors'].buttonDefaultColor : props.theme['colors'].background};
    transition: all 0.35s ease;

    &:hover {
        background-color: ${props => props.$isChecked ? props.theme['colors'].background : props.theme['colors'].buttonDefaultColor};
    }
`;

const CheckMark = styled(CheckSVG)`  
    color: ${props => props.theme['colors'].buttonDefaultColor};   
`;

type CheckboxProps = {
    name: string;
    index?: number;
    isChecked: boolean;
    width? :string;
    height?: string;
    onSelect: (index: number, newSelectState: boolean) => void;
};

const Checkbox: React.FC<CheckboxProps> = (props: CheckboxProps) => {
    const index = props.index ? props.index : 0;
    const width = props.width ? props.width : "20px";
    const height = props.height ? props.height : "20px";

    return (
        <CheckboxWrapper>
            <CheckboxInput id={`${props.name}_input`} type="checkbox" checked={props.isChecked} onChange={() => props.onSelect(index, !props.isChecked)} />
            <CheckboxLabel id={`${props.name}_label`} $width={width} $height={height} $isChecked={props.isChecked}>
                {props.isChecked && <CheckMark />}
            </CheckboxLabel>
        </CheckboxWrapper>
    );
};

export default Checkbox;
