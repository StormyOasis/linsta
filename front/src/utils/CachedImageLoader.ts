import { clear, del, get, set } from 'idb-keyval';
import { blobToBase64 } from "./utils";

export const loadImageCached = async (id: string, url: string): Promise<string> => {
    // First try to get the image from the cache
    const value = await get(id);

    if(value === undefined) {
        // Image not in the cache so have to fetch it
        try {
            const result = await fetch(url);
            if(result.status === 200) {
                const blob = await result.blob();
                const b64:string = await blobToBase64(blob) as string;

                // add the base64 data into the cache and return it
                await set(id, b64);

                return b64;
            }
        } catch(err) {
            console.log(err);
            return url; //default to what was passed in
        }

    } else {
        // Image is in the cache already as a base64 string, use that as is
        return value;
    }

    return url; //default to what was passed in
}

export const setImage = async (id: string, data: string) => {     
    await set(id, data);
}

export const deleteFromCache = async (id: string) => {
    // First try to get the image from the cache
    const value = await get(id);

    if(value === undefined) {
        // key not present in cache so do nothing
        return;
    }

    // delete from the cache
    await del(id);
}

export const clearCache = async () => {
    await clear();
}