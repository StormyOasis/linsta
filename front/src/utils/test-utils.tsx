import React, { ReactElement } from 'react';
import { render, RenderOptions } from '@testing-library/react';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import { RootState } from "../Components/Redux/redux";
import { authInitialState } from '../Components/Redux/slices/auth.slice';
import { modalInitialState } from '../Components/Redux/slices/modals.slice';
import { profileInitialState } from '../Components/Redux/slices/profile.slice';
import { miscInitialState } from '../Components/Redux/slices/misc.slice';
import { configureStore } from '@reduxjs/toolkit';

import Theme from '../Components/Themes/Theme';

export const mockInitialState = (overrides?: Partial<RootState>): RootState => ({
    auth: { ...authInitialState, ...(overrides?.auth ?? {}) },
    modal: { ...modalInitialState, ...(overrides?.modal ?? {}) },
    profile: { ...profileInitialState, ...(overrides?.profile ?? {}) },
    misc: { ...miscInitialState, ...(overrides?.misc ?? {}) },
});

type ExtendedRenderOptions = {
    store?: ReturnType<typeof configureStore>;
    preloadedState?: any;
    route?: string; // ‚Üê allow overriding the initial route
} & Omit<RenderOptions, "queries">;

const dummyAuthReducer = (state = { user: null, isLoading: false, error: null }, _action: any) => state;
const dummyModalReducer = (state = { isOpen: false }, _action: any) => state;
const dummyProfileReducer = (state = { name: '', email: '' }, _action: any) => state;
const dummyMiscReducer = (state = { someData: null }, _action: any) => state;

const renderWithStore = (
    ui: ReactElement,
    {
        preloadedState,
        store = configureStore({
            reducer: {
                auth: dummyAuthReducer,
                modal: dummyModalReducer,
                profile: dummyProfileReducer,
                misc: dummyMiscReducer,
            },
            preloadedState,
        }),
        route = "/",
        ...renderOptions
    }: ExtendedRenderOptions = {}
) => {
    const Wrapper: React.FC<{ children: React.ReactNode }> = ({ children }) => {
        return (
            <Provider store={store}>
                <MemoryRouter initialEntries={[route]}>
                    <Theme>{children}</Theme>
                </MemoryRouter>
            </Provider>
        );
    };

    return {
        ...render(ui, { wrapper: Wrapper, ...renderOptions }),
        store,
    };
};

export * from "@testing-library/react";
export { renderWithStore };
export type { ExtendedRenderOptions };