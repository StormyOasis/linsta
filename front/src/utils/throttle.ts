import { useRef } from 'react';

const useThrottle = (callback: Function, delay: number) => {
    const lastCall = useRef(0);

    return (...args: any[]) => {
        const now = new Date().getTime();
        if (now - lastCall.current >= delay) {
            lastCall.current = now;
            callback(...args);
        }
    };
};

export default useThrottle;