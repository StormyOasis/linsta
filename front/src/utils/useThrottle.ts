import { useRef, useCallback } from 'react';

/**
 * React hook for throttling a callback function.
 * The callback will only be invoked at most once every `delay` milliseconds.
 *
 * @param callback - The function to throttle.
 * @param delay - The minimum delay (ms) between calls.
 * @returns A throttled version of the callback.
 */
const useThrottle = <T extends (...args: any[]) => void>(callback: T, delay: number) => {
    const lastCall = useRef<number>(0);

    return useCallback(
        (...args: Parameters<T>) => {
            const now = performance.now();
            if (now - lastCall.current >= delay) {
                lastCall.current = now;
                callback(...args);
            }
        },
        [callback, delay]
    );
};

export default useThrottle;