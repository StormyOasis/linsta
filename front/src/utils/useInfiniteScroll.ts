import React, { useEffect, useRef } from 'react';
import useThrottle from './useThrottle';

/**
 * Type for the async load more function.
 */
type AsyncLoadMoreFunction = () => Promise<void>;

/**
 * Custom hook for infinite scrolling.
 * Attaches a throttled scroll event listener to a container and triggers `loadMore`
 * when the user scrolls near the bottom.
 *
 * @param loadMore - Async function to load more items.
 * @param isLoading - Whether a load is currently in progress.
 * @param containerRef - Ref to the scrollable container element.
 */
const useInfiniteScroll = (loadMore: AsyncLoadMoreFunction, 
    isLoading: boolean, 
    childRef: React.MutableRefObject<HTMLDivElement | null>) => {
        
    const hasScrolled = useRef<boolean>(false);  // Track if the user has scrolled

    // Throttled scroll handler
    const throttledHandleScroll = useThrottle(async () => {
        if (typeof window !== 'undefined' && childRef.current) {
            const element = childRef.current as HTMLElement;
            const currentScroll = window.innerHeight + element.scrollTop;

            if (currentScroll + 256 >= element.scrollHeight && !isLoading && !hasScrolled.current) {
                hasScrolled.current = true;  // Mark as scrolled at least once
                await loadMore();
            }
        }
    }, 200);

    useEffect(() => {
        const childElement = childRef.current;
        if (childElement) {
            childElement.addEventListener('scroll', throttledHandleScroll);
        }

        // Clean up event listener on unmount
        return () => {
            if (childElement) {
                childElement.removeEventListener('scroll', throttledHandleScroll);
            }
        };
    }, [throttledHandleScroll, childRef]);

    // Reset trigger flag when loading finishes
    useEffect(() => {
        if (!isLoading) {
            hasScrolled.current = false;
        }
    }, [isLoading]);
}

export default useInfiniteScroll;