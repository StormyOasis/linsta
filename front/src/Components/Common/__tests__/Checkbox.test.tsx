import React from "react";
import { render, screen, fireEvent } from "@testing-library/react";
import Checkbox from "../Checkbox";
import { renderWithStore } from "../../../utils/test-utils";
import '@testing-library/jest-dom';

describe("Checkbox", () => {
    const defaultProps = {
        name: "test",
        isChecked: false,
        onSelect: jest.fn(),
    };

    it("renders without crashing", () => {
        const r = renderWithStore(<Checkbox {...defaultProps} />);
        expect(screen.getByRole("checkbox")).toBeInTheDocument();
        expect(r).toMatchSnapshot();
    });

    it("shows checkmark when checked", () => {
        const r = renderWithStore(<Checkbox {...defaultProps} isChecked={true} />);
        // The checkmark is rendered as an SVG, so check for svg in the label
        const label = screen.getByTestId("checkbox-label");
        expect(label.querySelector("svg")).toBeInTheDocument();
        expect(r).toMatchSnapshot();
    });

    it("does not show checkmark when not checked", () => {
        const r = renderWithStore(<Checkbox {...defaultProps} isChecked={false} />);
        const label = screen.getByTestId("checkbox-label");
        expect(label.querySelector("svg")).not.toBeInTheDocument();
        expect(r).toMatchSnapshot();
    });

    it("calls onSelect with correct arguments when clicked", () => {
        const r = renderWithStore(<Checkbox {...defaultProps} index={2} />);
        const checkbox = screen.getByRole("checkbox");
        fireEvent.click(checkbox);
        expect(defaultProps.onSelect).toHaveBeenCalledWith(2, true);
        expect(r).toMatchSnapshot();
    });

    it("uses default width and height if not provided", () => {
        const r = renderWithStore(<Checkbox {...defaultProps} />);
        const label = screen.getByTestId("checkbox-label");
        expect(label).toHaveStyle("width: 20px");
        expect(label).toHaveStyle("height: 20px");
        expect(r).toMatchSnapshot();
    });

    it("applies custom width and height if provided", () => {
        const r = renderWithStore(<Checkbox {...defaultProps} width="30px" height="40px" />);
        const label = screen.getByTestId("checkbox-label");
        expect(label).toHaveStyle("width: 30px");
        expect(label).toHaveStyle("height: 40px");
        expect(r).toMatchSnapshot();
    });
});