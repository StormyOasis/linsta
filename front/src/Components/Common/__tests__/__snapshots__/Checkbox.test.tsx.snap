// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox applies custom width and height if provided 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="CombinedStyling__Div-sc-c3jdxk-0 CombinedStyling__Flex-sc-c3jdxk-2 Checkbox__CheckboxWrapper-sc-1apy48e-0 escqfF ckFvxY bIjEeY"
      >
        <input
          class="Checkbox__CheckboxInput-sc-1apy48e-1 dJbqHN"
          id="test_input"
          type="checkbox"
        />
        <label
          class="Checkbox__CheckboxLabel-sc-1apy48e-2 fvNAZc"
          data-testid="checkbox-label"
          id="test_label"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="CombinedStyling__Div-sc-c3jdxk-0 CombinedStyling__Flex-sc-c3jdxk-2 Checkbox__CheckboxWrapper-sc-1apy48e-0 escqfF ckFvxY bIjEeY"
    >
      <input
        class="Checkbox__CheckboxInput-sc-1apy48e-1 dJbqHN"
        id="test_input"
        type="checkbox"
      />
      <label
        class="Checkbox__CheckboxLabel-sc-1apy48e-2 fvNAZc"
        data-testid="checkbox-label"
        id="test_label"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;

exports[`Checkbox calls onSelect with correct arguments when clicked 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="CombinedStyling__Div-sc-c3jdxk-0 CombinedStyling__Flex-sc-c3jdxk-2 Checkbox__CheckboxWrapper-sc-1apy48e-0 escqfF ckFvxY bIjEeY"
      >
        <input
          class="Checkbox__CheckboxInput-sc-1apy48e-1 dJbqHN"
          id="test_input"
          type="checkbox"
        />
        <label
          class="Checkbox__CheckboxLabel-sc-1apy48e-2 gCSPEX"
          data-testid="checkbox-label"
          id="test_label"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="CombinedStyling__Div-sc-c3jdxk-0 CombinedStyling__Flex-sc-c3jdxk-2 Checkbox__CheckboxWrapper-sc-1apy48e-0 escqfF ckFvxY bIjEeY"
    >
      <input
        class="Checkbox__CheckboxInput-sc-1apy48e-1 dJbqHN"
        id="test_input"
        type="checkbox"
      />
      <label
        class="Checkbox__CheckboxLabel-sc-1apy48e-2 gCSPEX"
        data-testid="checkbox-label"
        id="test_label"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;

exports[`Checkbox does not show checkmark when not checked 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="CombinedStyling__Div-sc-c3jdxk-0 CombinedStyling__Flex-sc-c3jdxk-2 Checkbox__CheckboxWrapper-sc-1apy48e-0 escqfF ckFvxY bIjEeY"
      >
        <input
          class="Checkbox__CheckboxInput-sc-1apy48e-1 dJbqHN"
          id="test_input"
          type="checkbox"
        />
        <label
          class="Checkbox__CheckboxLabel-sc-1apy48e-2 gCSPEX"
          data-testid="checkbox-label"
          id="test_label"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="CombinedStyling__Div-sc-c3jdxk-0 CombinedStyling__Flex-sc-c3jdxk-2 Checkbox__CheckboxWrapper-sc-1apy48e-0 escqfF ckFvxY bIjEeY"
    >
      <input
        class="Checkbox__CheckboxInput-sc-1apy48e-1 dJbqHN"
        id="test_input"
        type="checkbox"
      />
      <label
        class="Checkbox__CheckboxLabel-sc-1apy48e-2 gCSPEX"
        data-testid="checkbox-label"
        id="test_label"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;

exports[`Checkbox renders without crashing 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="CombinedStyling__Div-sc-c3jdxk-0 CombinedStyling__Flex-sc-c3jdxk-2 Checkbox__CheckboxWrapper-sc-1apy48e-0 escqfF ckFvxY bIjEeY"
      >
        <input
          class="Checkbox__CheckboxInput-sc-1apy48e-1 dJbqHN"
          id="test_input"
          type="checkbox"
        />
        <label
          class="Checkbox__CheckboxLabel-sc-1apy48e-2 gCSPEX"
          data-testid="checkbox-label"
          id="test_label"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="CombinedStyling__Div-sc-c3jdxk-0 CombinedStyling__Flex-sc-c3jdxk-2 Checkbox__CheckboxWrapper-sc-1apy48e-0 escqfF ckFvxY bIjEeY"
    >
      <input
        class="Checkbox__CheckboxInput-sc-1apy48e-1 dJbqHN"
        id="test_input"
        type="checkbox"
      />
      <label
        class="Checkbox__CheckboxLabel-sc-1apy48e-2 gCSPEX"
        data-testid="checkbox-label"
        id="test_label"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;

exports[`Checkbox shows checkmark when checked 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="CombinedStyling__Div-sc-c3jdxk-0 CombinedStyling__Flex-sc-c3jdxk-2 Checkbox__CheckboxWrapper-sc-1apy48e-0 escqfF ckFvxY bIjEeY"
      >
        <input
          checked=""
          class="Checkbox__CheckboxInput-sc-1apy48e-1 dJbqHN"
          id="test_input"
          type="checkbox"
        />
        <label
          class="Checkbox__CheckboxLabel-sc-1apy48e-2 jDynnf"
          data-testid="checkbox-label"
          id="test_label"
        >
          <svg
            aria-hidden="true"
            class="Icon-sc-btq2kp-0 Checkbox__CheckMark-sc-1apy48e-3 liWlFw"
            fill="none"
            focusable="false"
            name="check"
            stroke="currentColor"
            stroke-width="2"
          >
            <use
              href="http://localhost:8080/public/images/sprite.svg#check"
            />
          </svg>
        </label>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="CombinedStyling__Div-sc-c3jdxk-0 CombinedStyling__Flex-sc-c3jdxk-2 Checkbox__CheckboxWrapper-sc-1apy48e-0 escqfF ckFvxY bIjEeY"
    >
      <input
        checked=""
        class="Checkbox__CheckboxInput-sc-1apy48e-1 dJbqHN"
        id="test_input"
        type="checkbox"
      />
      <label
        class="Checkbox__CheckboxLabel-sc-1apy48e-2 jDynnf"
        data-testid="checkbox-label"
        id="test_label"
      >
        <svg
          aria-hidden="true"
          class="Icon-sc-btq2kp-0 Checkbox__CheckMark-sc-1apy48e-3 liWlFw"
          fill="none"
          focusable="false"
          name="check"
          stroke="currentColor"
          stroke-width="2"
        >
          <use
            href="http://localhost:8080/public/images/sprite.svg#check"
          />
        </svg>
      </label>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;

exports[`Checkbox uses default width and height if not provided 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="CombinedStyling__Div-sc-c3jdxk-0 CombinedStyling__Flex-sc-c3jdxk-2 Checkbox__CheckboxWrapper-sc-1apy48e-0 escqfF ckFvxY bIjEeY"
      >
        <input
          class="Checkbox__CheckboxInput-sc-1apy48e-1 dJbqHN"
          id="test_input"
          type="checkbox"
        />
        <label
          class="Checkbox__CheckboxLabel-sc-1apy48e-2 gCSPEX"
          data-testid="checkbox-label"
          id="test_label"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="CombinedStyling__Div-sc-c3jdxk-0 CombinedStyling__Flex-sc-c3jdxk-2 Checkbox__CheckboxWrapper-sc-1apy48e-0 escqfF ckFvxY bIjEeY"
    >
      <input
        class="Checkbox__CheckboxInput-sc-1apy48e-1 dJbqHN"
        id="test_input"
        type="checkbox"
      />
      <label
        class="Checkbox__CheckboxLabel-sc-1apy48e-2 gCSPEX"
        data-testid="checkbox-label"
        id="test_label"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;
